content/blog/2022/cq-syntax.md

```diff
@@ -0,0 +1,396 @@
+any breaking changes.
+{% endcallout %}
+
+Container Queries
+allow us to measure (or 'query')
+aspects of a 'container' element,
+and style any descendants
+based on the result of the query.
+This is very similar to
+Media Queries,
+which allow us to query the overall
+_browser viewport_.
+
+If you haven't encountered
+Container Queries before,
+there are many
+[resources available](https://css.oddbird.net/rwd/query/#articles--demos)
+on the web.
+[David Herron](/authors/davidh/)
+wrote a great
+[Quick Start Guide](/2021/04/05/containerqueries/)
+back in April 2021,
+when Chrome Canary
+shipped the first prototype of the feature.
+
+Since then
+the syntax has gone through several revisions.
+That's why we prototype --
+it allows us to learn and make changes,
+before developers are relying on the feature
+in production.
+The downside is,
+many people (including me!) have written articles,
+made demos,
+given talks,
+and released videos
+using now-out-of-date syntax.
+
+Now Safari and Chrome
+have both signaled that
+they are ready to ship
+Container size queries and units,
+likely starting in late August 2022.
+
+So, what syntax
+is actually shipping in browsers?
+
+## TL;DR, a code sample
+
+```css
+main, .sidebar {
+  /* establish containers for inline-size queries */
+  container-type: inline-size;
+}
+
+main {
+  /* optionally give a container one or more names */
+  /* - use any names you want, there are no pre-defined options */
+  container-name: main-content page-layout;
+}
+
+article {
+  /* a shorthand to set both names and types */
+  /* - names are required and come first in the shorthand */
+  container: article-layout / inline-size;
+}
+
+/* query the nearest ancestor container named page-layout */
+/* - using a container-name is optional in the query */
+/* - we can also use the old `(min-width: 40em)` syntax */
+@container page-layout (inline-size > 40em) {
+  .card {
+    grid-template-columns: auto 1fr;
+  }
+}
+
+h2 {
+  /* container-query-inline units */
+  font-size: max(1.2em, 1em + 2cqi);
+}
+```
+
+If you are looking at an old demo,
+talk, or article,
+and wondering why it doesn't work,
+there are two major changes
+that you should check first:
+
+- [The container shorthand](#the-container-shorthand)
+- [Container query units](#container-query-units)
+
+Let's dig into the details...
+
+## Establishing containers with `container-type`
+
+In order to query elements in the page,
+we need to define them as _query containers_.
+We can do that with the `container-type` property,
+which defines the questions we're able to ask
+about a given container.
+
+The initial value is `normal`.
+By default,
+all elements are _style containers_ --
+meaning we should be able to query the
+[computed value](https://developer.mozilla.org/en-US/docs/Web/CSS/computed_value)
+of any property on any element.
+But don't worry about that yet,
+**browsers are only shipping size queries in September 2022**.[^style]
```

Comment:
Everywhere else we mention "late August" or "August 30", so I found this a bit confusing.

Reply:
Oops, yeah - pushed a fix